File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, timePerLayer int) int {
    if timePerLayer == 0 {
        timePerLayer = 2
    }
    return timePerLayer * len(layers)
}

// TODO: define the 'Quantities()' function
func Quantities(layers []string) (noodles int, sauce float64) {
    for _, layer := range(layers) {
        if layer == "sauce" {
            sauce += 0.2
        } else if layer == "noodles" {
            noodles += 50
        }
    }
    return
}


// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friendsList []string, myList []string) {
    myList[len(myList)-1] = friendsList[len(friendsList)-1]
    
}

// TODO: define the 'ScaleRecipe()' 
func ScaleRecipe(quantities []float64, portions int) []float64 {
    scaledQuantities := make([]float64, len(quantities))
    for i, quantity := range(quantities) {
        scaledQuantities[i] = (quantity * float64(portions))/2
    }
    return scaledQuantities
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
// 
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more 
// functionality.

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */
export function cookingStatus(timer) {
  if (timer === undefined) {
    return 'You forgot to set the timer.'
  }
  if (timer === 0) {
    return 'Lasagna is done.'
  }
  return 'Not done, please wait.'
}

export function preparationTime(layers, avgTimePerLayer=2) {
  return avgTimePerLayer*layers.length;
}

export function quantities(layers) {
  let quant = {
    "noodles": 0,
    "sauce": 0
  };
  let quantitiesByLayer = {
    "noodles": 50,
    "sauce": 0.2
  };
  for (let layer of layers) {
    if (!Object.hasOwn(quantitiesByLayer, layer)) {
      continue;
    }
    quant[layer] += quantitiesByLayer[layer];
  }
  return quant;
}

export function addSecretIngredient(friendsList, myList) {
  myList.push(friendsList[friendsList.length-1]);
}

export function scaleRecipe(recipe, portions) {
  let scalingFactor = portions/2;
  let scaledRecipe = {...recipe};
  for (let ingredient in recipe) {
    scaledRecipe[ingredient] = recipe[ingredient] * scalingFactor;
  }
  return scaledRecipe;
  
}
